design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -75,21 +81,30 @@ It would be more consistent with other authenticator configs if authn-oidc were
 
 
 ## Implementation
-An existing workaround for the CA cert issue mentioned previously involves a user manually updating the OpenSSL truststore in the container to include any custom CA certs. We can leverage this idea in Conjur to do the same thing on the fly. A simple wrapper method which creates a temporary truststore in the container and sets the OpenSSL environment variable `SSL_CERT_FILE` to point to this tempfile should be sufficient. Cleanup involves unsetting or resetting the environment variable to its original value, and ensuring that the tempfile has been cleaned up after code execution.
+An existing workaround for the CA cert issue mentioned previously involves a user manually updating the 
+OpenSSL truststore in the container to include any custom CA certs. We can leverage this idea in Conjur 
+to do the same thing on the fly. A simple wrapper method which creates a temporary symlink in the OpenSSL 
+truststore to a tempfile containing the certificate content should be sufficient. Cleanup involves 
+removing the symlink, and ensuring that the tempfile has been cleaned up after code execution. Since we 
+are (temporarily) adding the cert to the existing truststore, it should not interfere with concurrent 
+HTTP connections which may rely on the default OpenSSL certs still being available.
 
 An example of what this wrapper method may look like:
 ```ruby
 def with_temporary_cert_store(cert_string)
```

Comment:
We can sort it out later, but this approach won't work for CA certs that are chains (the certificate creation will fail).  For those, we need to write the cert to a temp file and import it (example I previously posted is part of the authn-jwt rewrite, which includes support for CA chains).

Reply:
There's already a [parse_certs method](https://github.com/cyberark/conjur/blob/322861bbc8897cd6dc2be0d50b81b6a678024473/app/domain/conjur/cert_utils.rb#L11) in the CertUtils module which I imagine can be repurposed to do exactly what you're describing. But yes we would need slightly modified logic to create separate temp files/symlinks for each cert that gets parsed from the chain

Reply:
Really just making sure our eventual tests include examples with the configured CA certificate being a single certificate as well as a certificate chain.